Vineyard/Vineyard/Preview Content/VineBranchView.swift

```diff
@@ -0,0 +1,128 @@
+
+//
+//  VineBranchView.swift
+//  Vineyard
+//
+//  Created by Jiyoon Lee on 11/5/24.
+//
+
+import SwiftUI
+
+struct VineBranchView: View {
+    @State private var hasScrolledToEnd: Bool = false
+    @State private var hasScrolledToStart: Bool = true
+    var isStaticPreview: Bool = false
+    var body: some View {
+        ZStack {
+            if isStaticPreview {
+                staticImagePreview()
+            } else {
+                EndDetectionScrollView(.horizontal, showIndicators: false, hasScrolledToEnd: $hasScrolledToEnd, hasScrolledToStart: $hasScrolledToStart) {
+                    LazyHStack {
+                        ForEach(1..<11, id: \.self) { index in
```

Comment:
This shouldn't be a fixed value of 11 but instead the length of the array of vines that the user has